.travis.yml

```diff
@@ -3,18 +3,42 @@ language: python
 python: "2.7"
 
 env:
-  - ANSIBLE_ROLES_PATH="~/.ansible/roles" TRAVIS_TEST=1
+  global:
+    - ANSIBLE_ROLES_PATH="~/.ansible/roles"
+    - TRAVIS_TEST=1
+  matrix:
+    - TEST=main
+    - TEST=prove-deploy
+
+matrix:
+  fast_finish: true
+  allow_failures:
+    - env: TEST=prove-deploy
 
 before_install:
+  - if [[ $TEST = 'prove-deploy' ]] && git log --pretty=format:'%s %b' -n1 HEAD^2 | grep -iv '\[pv-deploy\]'; then exit 0; fi
```

Comment:
What is this line doing?

Reply:
Yeah that's pretty unreadable, I can add in a comment, but It's only running the prove-deploy test if the commit message contains `[pv-deploy]`:

`if [[ $TEST = 'prove-deploy' ]]`: if the test being run is `prove-deploy`
`&& git log --pretty=format:'%s %b' -n1 HEAD^2` and the commit message of the commit that was merged in to master
`| grep -iv '\[pv-deploy\]'`: doesn't contain the pattern `[pv-deploy]`
`; then exit 0; fi` then exit out of here